lib/cisco_node_utils/ace.rb

```diff
@@ -99,22 +108,53 @@ def ace_set(attrs)
 
       if attrs[:remark]
         cmd = 'ace_remark'
+        set_args_keys(attrs)
       else
         cmd = 'ace'
+        set_args_keys_default
+        set_args_keys(attrs)
         [:action,
          :proto,
          :src_addr,
          :src_port,
          :dst_addr,
          :dst_port,
+         :tcp_flags,
+         :established,
+         :precedence,
+         :dscp,
+         :time_range,
+         :packet_length,
+         :ttl,
+         :http_method,
+         :tcp_option_length,
+         :redirect,
+         :log,
         ].each do |p|
           attrs[p] = '' if attrs[p].nil?
+          send(p.to_s + '=', attrs[p])
         end
+        @get_args = @set_args
       end
-      set_args_keys(attrs)
       config_set('acl', cmd, @set_args)
     end
 
+    # UTILITY FUNCTIONS
+    # -----------------
+    def extract_value(prop, prefix)
```

Comment:
In most of these cases `prop` is the same as `prefix` so you could just make `prefix` optional to simplify the callers:

```
  def extract_value(prop, prefix=nil)
    prefix = prop if prefix.nil?
```